### DISCUSSION TITLE
Gaps in wms layer on high zoom levels

### DISCUSSION THREAD

**Discussion by AndiLeni** [openlayers/openlayers#15899] [Q&A] [answered]
Hello,

I have a WMS layer which makes these kind of gaps a high zoom levels.
Is this related to openlayers or an issue caused by my WMS provider?

Thanks and kind regards

![image](https://github.com/openlayers/openlayers/assets/13385771/8f1d2b71-b3ed-4a8a-bf54-3674e1bcc1a9)

**Comment by ahocevar** [comment]
Looks like an issue with your WMS provider. In the network tab of your dev tools, find the requests that get sent to the WMS, and see how the images look in there. Also check the request URLs to see if they are plausible.

**Reply by AndiLeni** [reply]
Thanks for your answer and sorry for the late reply. 
The urls look like this: `https://maps.dwd.de/geoserver/dwd/wms?REQUEST=GetMap&SERVICE=WMS&VERSION=1.3.0&FORMAT=image%2Fpng&STYLES=&TRANSPARENT=true&LAYERS=dwd%3ANiederschlagsradar&WIDTH=256&HEIGHT=256&CRS=EPSG%3A3857&BBOX=1198532.6035115644%2C6171219.915631991%2C1200978.58841669%2C6173665.900537116`

Which results in this image (at the time or writing):
![image](https://github.com/user-attachments/assets/6fa71417-2112-4050-b47b-d60828246043)

I am not really experienced with mapping yet, does this url look good to you?

**Comment by ahocevar** [comment, marked as answer]
There seems to be an issue with the server. Try the following layer configuration to avoid requesting zooms that are too high:
```js
new TileLayer({
    source: new TileWMS({
      tileGrid: createXYZ({
        tileSize: 512,
        maxZoom: 12,
      }),
      url: 'https://maps.dwd.de/geoserver/dwd/wms',
      params: {LAYERS: 'dwd:Niederschlagsradar', TILED: true},
      transition: 0,
    }),
  }),
```
This restricts requests to zoom 12 at maximum. If you zoom in further the map will still display the layer, but it will use lower zoom tiles.

Here is a map with that configuration live: https://3r66jg.csb.app

**Reply by AndiLeni** [reply]
Thank you very much! This is working fine currently :-)
I will report this issue to the DWD and see what they say about this.

**Reply by AndiLeni** [reply]
@ahocevar The Layer seems to use a custom projection: EPSG:1000001 which seems to cause the splits in high zoom levels.
I am not sure how this projection might be used with openlayers, so I think I will stick with the maxZoom method.

**Reply by ahocevar** [reply]
@AndiLeni If you can get a projection definition, you can make OpenLayers use it. If you have a WKT or proj definition then with proj4, if you get forward and inverse transform functions, then you can register them directly with OpenLayers.
